src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -13,136 +14,149 @@ internal class JsonSchemaTransformer
     {
         private readonly JsonSchema _schema;
         private readonly JsonSchemaDefinition _definitions;
+        private readonly ConcurrentDictionary<Document, Dictionary<JToken, JToken>> _transformedProperties;
 
         public JsonSchemaTransformer(JsonSchema schema)
         {
             _schema = schema;
             _definitions = new JsonSchemaDefinition(schema);
+            _transformedProperties = new ConcurrentDictionary<Document, Dictionary<JToken, JToken>>();
         }
 
         public (List<Error> errors, JToken token) TransformContent(Document file, Context context, JToken token)
         {
             var errors = new List<Error>();
-            var transformedToken = Transform(file, context, _schema, token, errors);
+            var transformedToken = TransformToken(file, context, _schema, token, errors);
             return (errors, transformedToken);
         }
 
-        public (List<Error> errors, Dictionary<string, List<(bool isRoot, SourceInfo source, Dictionary<string, Lazy<JToken>> propertiesByUid)>> xrefPropertiesGroupByUid) TransformXref(Document file, Context context, JToken token)
+        public (List<Error> errors, Dictionary<string, List<(bool isRoot, SourceInfo source, Dictionary<string, Lazy<JToken>> propertiesByUid)>> xrefPropertiesGroupByUid) TraverseXref(Document file, Context context, JToken token)
```

Comment:
can simplify the interface to return a list of `InternalXrefSpec`

```suggestion
        public (List<Error> errors, List<InternalXrefSpec> xrefs)>) LoadXrefSpecs(Document file, Context context, JToken token)
```

Reply:
I think we can do this in another PR, since it need change XrefMapBuild also, which is not this PR focus.